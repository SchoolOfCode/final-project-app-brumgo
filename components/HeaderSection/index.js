import React, { Component } from "react";
import { Header } from "react-native-elements";

export default class HeaderSection extends Component {
  render() {
    return (
      <Header
        statusBarProps={{ barStyle: "light-content" }}
        barStyle="light-content"
        leftComponent={{ icon: "menu", color: "#fff" }}
        centerComponent={{ text: "brumGo", style: { color: "#fff" } }}
        rightComponent={{ icon: "home", color: "#fff" }}
        containerStyle={{
          backgroundColor: "#E12B38",
          justifyContent: "space-around"
        }}
      />
    );
  }
}
