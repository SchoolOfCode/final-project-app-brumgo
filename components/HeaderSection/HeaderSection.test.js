import React from "react";
import * as renderer from "react-test-renderer";
import HeaderSection from "./index";

describe("Header Section", () => {
  it("should match to snapshot", () => {
    const component = renderer.create(<HeaderSection />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
