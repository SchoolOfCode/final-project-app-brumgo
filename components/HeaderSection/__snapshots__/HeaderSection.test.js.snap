// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Section should match to snapshot 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "alignItems": "flex-start",
      "backgroundColor": "#E12B38",
      "borderBottomColor": "#f2f2f2",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "height": "8%",
      "justifyContent": "space-around",
      "paddingHorizontal": 10,
      "paddingTop": 12.5,
    }
  }
>
  <Image
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": "8%",
          "width": undefined,
        },
        undefined,
      ]
    }
    testID="headerContainer"
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
        }
      }
    >
      <View
        icon="menu"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
        underlayColor="white"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 24,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 3,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#fff",
        }
      }
      text="brumGo"
      theme={
        Object {
          "colors": Object {
            "disabled": "hsl(208, 8%, 90%)",
            "divider": "#bcbbc1",
            "error": "#ff190c",
            "grey0": "#393e42",
            "grey1": "#43484d",
            "grey2": "#5e6977",
            "grey3": "#86939e",
            "grey4": "#bdc6cf",
            "grey5": "#e1e8ee",
            "greyOutline": "#bbb",
            "platform": Object {
              "android": Object {
                "error": "#f44336",
                "primary": "#2196f3",
                "secondary": "#9C27B0",
                "success": "#4caf50",
                "warning": "#ffeb3b",
              },
              "ios": Object {
                "error": "#ff3b30",
                "primary": "#007aff",
                "secondary": "#5856d6",
                "success": "#4cd964",
                "warning": "#ffcc00",
              },
            },
            "primary": "#2089dc",
            "searchBg": "#303337",
            "secondary": "#8F0CE8",
            "success": "#52c41a",
            "warning": "#faad14",
          },
        }
      }
    >
      brumGo
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
        }
      }
    >
      <View
        icon="home"
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
        theme={
          Object {
            "colors": Object {
              "disabled": "hsl(208, 8%, 90%)",
              "divider": "#bcbbc1",
              "error": "#ff190c",
              "grey0": "#393e42",
              "grey1": "#43484d",
              "grey2": "#5e6977",
              "grey3": "#86939e",
              "grey4": "#bdc6cf",
              "grey5": "#e1e8ee",
              "greyOutline": "#bbb",
              "platform": Object {
                "android": Object {
                  "error": "#f44336",
                  "primary": "#2196f3",
                  "secondary": "#9C27B0",
                  "success": "#4caf50",
                  "warning": "#ffeb3b",
                },
                "ios": Object {
                  "error": "#ff3b30",
                  "primary": "#007aff",
                  "secondary": "#5856d6",
                  "success": "#4cd964",
                  "warning": "#ffcc00",
                },
              },
              "primary": "#2089dc",
              "searchBg": "#303337",
              "secondary": "#8F0CE8",
              "success": "#52c41a",
              "warning": "#faad14",
            },
          }
        }
        underlayColor="white"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#fff",
                "fontSize": 24,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
